tests/integration/models/device-type.spec.ts

```diff
@@ -159,7 +160,10 @@ describe('Device Type model', function () {
 					{ $select: 'contract' },
 				);
 				assertExists(contract);
-				const result = await balena.models.deviceType.getInstructions(contract);
+				// @ts-expect-error - parsed contract will be a Contract
+				const $contract = contract as Contract;
```

Comment:
This is going to become super frustrating while using these methods....
Maybe we should make `getInstructions` access `Contract | typeof DeviceType['Read']['contract']`  just to make it a bit easier, and have it throw in case it doesn't look like a Contract.